app/src/main/java/com/sdp13epfl2021/projmag/activities/ProjectsListActivity.kt

```diff
@@ -81,6 +86,28 @@ class ProjectsListActivity : AppCompatActivity() {
             fab.visibility = View.INVISIBLE
         }
 
+        candidatureDatabase = utils.candidatureDatabase
+
+        appliedProjects.forEach{
+            utils.candidatureDatabase.addListener(it) { _: ProjectId, list : List<Candidature> ->
+                val candidatureThatChanged : Candidature? = list.find { candidature -> candidature.userId == utils.auth.currentUser?.uid }
+                if(candidatureThatChanged != null) {
+                    if(candidatureThatChanged.state == Candidature.State.Accepted) {
+                        val otherCandidatures = appliedProjects.filter { projectId -> (candidatureThatChanged.projectId != projectId) }
+                        otherCandidatures.forEach { otherCandidatureId ->
+                            candidatureDatabase.removeCandidature(
+                                    otherCandidatureId,
+                                    utils.auth.currentUser?.uid!!,
+                                    {},
+                                    {}
+                            )
+                            utils.userdataDatabase.applyUnapply(false, otherCandidatureId, {}, {})
+                            Log.d("MYTEST", "Removed a project")
```

Comment:
Log/print should be remove before adding to main

Reply:
Indeed